.changeset/moody-toys-dance.md

```diff
@@ -0,0 +1,31 @@
+---
+"@fluidframework/tree": minor
+"fluid-framework": minor
+---
+
+Allow constructing ArrayNodes from Maps and MapNodes from arrays when unambiguous.
+
+Since the types for ArrayNodes and MapNodes indicate they can be constructed from iterables,
+it should work, even if those iterables are themselves arrays or maps.
+To avoid this being a breaking change, a priority systems was introduced.
+ArrayNodes will only be implicitly constructable from JavaScript Map objects in contexts where no MapNodes are allowed.
+Similarly MapNodes will only be implicitly constructable from JavaScript Array objects in contexts where no ArrayNodes are allowed.
+
+In practice, the main case this is likely to matter is when implicitly constructing a map node, if you provide an array of key value pairs, this now works instead of erroring, as long as no ArrayNode is valid at that location in the tree.
```

Comment:
Nit
```suggestion
In practice, the main case in which this is likely to matter is when implicitly constructing a map node. If you provide an array of key value pairs, this now works instead of erroring, as long as no ArrayNode is valid at that location in the tree.
```